### PR TITLE
drush aliases, ssh configs and ds binary update commands

### PR DESCRIPTION
**Pull request by desmondmorris**
tldr; `ds update-db` will pull down the staging db and `ds update-files` will pull down the staging files

This merge introduces the managed drush alias file in the vagrant VM.  It gives us access to the app at the new staging environment (staging.beta).  To make this work, file ownership of the known_hosts fie needed to be granted to the vagrant user.  I also added a ssh to avoid strict host checking when ssh-ing into the staging instance.  This allows us to run drush commands against the staging instance directly from or VM.

```
drush @ds.staging cc all
drush @ds.staging fra
```

This also allows us to run rsync and sql-sync commands to quickly grab files and db from the staging instance. 

This rsync's the files from the staging instance down to the current environment.

```
drush rsync @ds.staging @self -y 
```

This pulls the db from the staging instance down to the current environment.

```
drush sql-sync @ds.staging @self -y
```

Both of these must be run from within a working drupal webroot, so /vagrant/html.

**WARNING** - These commands can be dangerous if you are not careful.  The first alias always represents the env you would like to pull from.  If you switch these up, you will bork the staging site.  So I created two convienience functions in the ds binary: `update-db` and `update-files`

**UPDATE**\* - I was able to add a drush policy file (go figure), to restrict the sync commands from overwriting the staging db and files

Do not disappoint dries
dries-buytaert_pan_23415

Related to: #675

### CODE REVIEW

**Comment by aaronschachter** [comment]
The dangerous part is scary!  What would be borked?

**Comment by desmondmorris** [comment]
@aaronschachter you could write over the staging db or files directories

**Comment by mshmsh5000** [comment]
@desmondmorris The drush alias should be using read-only credentials for DB. That would make overwriting the staging DB impossible.

**Comment by desmondmorris** [comment]
@mshmsh5000 +1 to that.  I realized I did not have to set the db credents at all in the alias file, so I did not.  But then I lose that granularity.  This def makes sense for now as staging is the canonical source of data at this point. 

The only challenge here is securing the mysql connection.  Unlike blackangus, this instance is available to the public.  So a passwordless login would not be ideal.  The alternative is to set the db username and password in code, also not ideal.  Is there anyway to restrict access to that mysql instance considering we are hitting it from a VM and at times distributed?  Maybe restrict to internal ip addresses, which require connecting to the VPN?

**Comment by mshmsh5000** [comment]
_Restrict to internal IP addresses_: +1. VPN should be the answer in general to things like this. I _assume_ this will work from within Vagrant, but haven't tested.

**Comment by desmondmorris** [comment]
Policy file added.  You will now be incapable of running sql-sync or rsync to the staging environment

File: provision/salt/roots/salt/top.sls [hunk 1]
```diff
@@ -7,4 +7,5 @@ base:
     - install
     - composer
     - node
+    - dosomething
```
Comment (aaronschachter) [inline provision/salt/roots/salt/top.sls:10]: what is this?

Comment (desmondmorris) [inline provision/salt/roots/salt/top.sls:10]: that tells salt to load the dosomething.sls file

**Comment by aaronschachter** [comment]
:+1:
